File: py_solutions_1-10/Euler_10.py
# Find the sum of all the primes below two million. 

from math import sqrt
import timeit


start = timeit.default_timer()

def is_prime(n):
    if n == 2:
        return True
    if n == 1 or n % 2 == 0:
        return False
    for i in range(3, int(sqrt(n) + 1), 2):
        if n % i == 0:
            return False
    return True

def prime_gen_under_two_mil():
    n = 0
    while n < 2000000:
        if n % 2 != 0 and is_prime(n):
            yield n
        n += 1

def euler_10():
    return sum([i for i in prime_gen_under_two_mil()])

print 'Answer: %s' % euler_10()
stop = timeit.default_timer()
print 'Time: %f' % (stop - start)

File: Java/java_solutions_1-10/Euler_10.java
/*
/   Project Euler problem #10 Primes below two million.
/
/   Find the sum of all the primes below two million.
/
*/

import static java.lang.Math.sqrt;


public class Euler_10
{
    public static boolean isPrime(int n)
    {
        if (n < 2) return false;
        else if (n == 2) return true;

        for (int i = 2; i < sqrt(n) + 1; i++) 
            if (n % i == 0) return false;

        return true;
    }

    public static void main(String[] args)
    {
        long start = System.nanoTime();
        long primeSum = 0;
        for (int i = 1; i < 2000000; i++) {
            if (!isPrime(i)) continue;
            primeSum += i;
        }

        System.out.println("Answer: " + primeSum);
        long stop = System.nanoTime();
        System.out.printf("Time: %.4f\n", ((float) stop - start) / 1_000_000_000);
    }
}
